test/e2e/node_feature_discovery_test.go

```diff
@@ -909,6 +910,74 @@ denyLabelNs: []
 					)).NotTo(HaveOccurred())
 				})
 			})
+
+			Context("and test whether resyncPeriod is passed successfully or not", func() {
+				BeforeEach(func(ctx context.Context) {
+					extraMasterPodSpecOpts = []testpod.SpecOption{
+						testpod.SpecWithConfigMap("nfd-master-conf", "/etc/kubernetes/node-feature-discovery"),
+					}
+					cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+resyncPeriod: "1s"
+`)
+					_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, cm, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+				It("labels should be restored to the original ones", func(ctx context.Context) {
+					// deploy node feature object
+					if !useNodeFeatureApi {
+						Skip("NodeFeature API not enabled")
+					}
+
+					nodes, err := getNonControlPlaneNodes(ctx, f.ClientSet)
+					Expect(err).NotTo(HaveOccurred())
+
+					targetNodeName := nodes[0].Name
+					Expect(targetNodeName).ToNot(BeEmpty(), "No suitable worker node found")
+
+					// Apply Node Feature object
+					By("Creating NodeFeature object")
+					nodeFeatures, err := testutils.CreateOrUpdateNodeFeaturesFromFile(ctx, nfdClient, "nodefeature-1.yaml", f.Namespace.Name, targetNodeName)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying node labels from NodeFeature object #1")
+					expectedLabels := map[string]k8sLabels{
+						targetNodeName: {
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-1": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-2": "obj-1",
+							nfdv1alpha1.FeatureLabelNs + "/fake-fakefeature3":      "overridden",
+						},
+					}
+					Expect(checkForNodeLabels(ctx, f.ClientSet,
+						expectedLabels, nodes,
+					)).NotTo(HaveOccurred())
+
+					k8sHelper := apihelper.K8sHelpers{
+						Kubeconfig: f.ClientConfig(),
+					}
+					k8sClient, err := k8sHelper.GetClient()
+					Expect(err).NotTo(HaveOccurred())
+
+					nodeToUpdate := getNode(nodes, targetNodeName)
+					err = k8sHelper.PatchNode(k8sClient, nodeToUpdate.Name, []apihelper.JsonPatch{
+						{
+							Op:    "replace",
+							Value: "randomValue",
+							Path:  "/metadata/labels/" + nfdv1alpha1.FeatureLabelNs + "/e2e-nodefeature-test-1",
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
The apihelper type is pretty useless here. Just use the available clientset directly. Also, use NewJsonPatch helper to get correctly formatted patch that applies to the node

```suggestion
                                        data, err := json.Marshal(apihelper.NewJsonPatch("replace", "/metadata/labels/", nfdv1alpha1.Featur
eLabelNs+"/e2e-nodefeature-test-1", "randomValue"))
                                        Expect(err).NotTo(HaveOccurred())
                                        _, err = f.ClientSet.CoreV1().Nodes().Patch(ctx, targetNodeName, types.JSONPatchType, data, metav1.
PatchOptions{})
                                        Expect(err).NotTo(HaveOccurred())

```